### PR TITLE
[Banner] Add snapshot tests for setting preferredFont with adjustsFontForContentSizeCategory

### LINKED ISSUE #8626: [Banner] Internal issue: b/143160672
This was filed as an internal issue. If you are a Googler, please visit [b/143160672](http://b/143160672) for more details.

---
#### Internal data
- Associated internal bug: [b/143160672](http://b/143160672)

### PR DESCRIPTION
**Pull request by yarneo**
This PR adds 2 snapshot tests to verify the behavior for setting preferredFont on the Banner's text elements when adjustsFontForContentSizeCategory is set to YES.

Closes #8626

### CODE REVIEW

**Review by jverkoey — COMMENTED** [review]
Are these tests surfacing a bug? Those screenshots look strange.

**Comment by yarneo** [comment]
Yeah, I believe this is the related bug: https://github.com/material-components/material-components-ios/issues/8326 but is out of scope in supporting dynamic type

**Comment by yarneo** [comment]
Just spoke to @wenyuzhang666 offline and he is aware of this and there are open bugs for the issue.

File: components/Banner/tests/snapshot/MDCBannerSnapshotTests.m [hunk 1]
```diff
@@ -437,4 +438,72 @@ - (void)testDynamicTypeForAttributedTextStringWhenContentSizeCategoryIsExtraExtr
...
+  if (@available(iOS 11.0, *)) {
+    // Given
+    self.bannerView = [[MDCBannerView alloc] init];
+    [self.bannerView applyThemeWithScheme:[[MDCContainerScheme alloc] init]];
+    self.bannerView.textView.text = kBannerLongText;
+    MDCButton *button1 = self.bannerView.leadingButton;
+    [button1 setTitle:@"Action1" forState:UIControlStateNormal];
+    [button1 setTitleColor:UIColor.blackColor forState:UIControlStateNormal];
+    button1.uppercaseTitle = YES;
+    button1.adjustsFontForContentSizeCategoryWhenScaledFontIsUnavailable = NO;
```
Comment (romoore) [inline components/Banner/tests/snapshot/MDCBannerSnapshotTests.m]: Let's chat when you get to the office. I'm surprised that this was necessary.  I'd also suggest setting `button.enableTitleFontForState = NO` so the `titleLabel.font` isn't replaced.

Comment (yarneo) [reply components/Banner/tests/snapshot/MDCBannerSnapshotTests.m]: Done

**Review by romoore — APPROVED** [review]

**Pull request merged**